src/main/sensors/barometer.c

```diff
@@ -174,11 +174,18 @@ int32_t baroCalculateAltitude(void)
 
 void performBaroCalibrationCycle(void)
 {
+    static int32_t savedGroundPressure=0;
+
     baroGroundPressure -= baroGroundPressure / 8;
     baroGroundPressure += baroPressureSum / PRESSURE_SAMPLE_COUNT;
     baroGroundAltitude = (1.0f - powf((baroGroundPressure / 8) / 101325.0f, 0.190295f)) * 4433000.0f;
 
-    calibratingB--;
+    if (baroGroundPressure==savedGroundPressure)
```

Comment:
I might add that it terminates after about 22 cycles indoors where there is no wind. With the new guesstimated pressure value one should consider changing CALIBRATING_BARO_CYCLES from 200 to e.g. 50.